sycl/doc/extensions/experimental/sycl_ext_oneapi_bindless_images.asciidoc

```diff
@@ -1001,8 +1001,16 @@ void write_image(unsampled_image_handle &ImageHandle,
 Inside a kernel, it's possible to read an image via `read_image`, passing 
 the image handle. For the form that takes `unsampled_image_handle`, image data 
 will be fetched exactly as is in device memory. For the form that takes a 
-`sampled_image_handle`, the image will be sampled according to the 
+`sampled_image_handle`, the image data will either be sampled or fetched, 
+depending on the type of `CoordT` passed. If the coordinates are 
+floating-point, the image will be sampled according to the 
 `bindless_image_sampler` that was passed to the image upon construction.
+If the coordinates are integers, the image data will be fetched, and no 
+sampling operations will be performed, the `bindless_image_sampler` passed to 
+the image upon creation has no effect when fetching sampled image data with 
+integer coordinates. Note that not all devices may support fetching of sampled 
+image data depending on the dimension or backing memory type. We provide 
+device aspect queries for this in <<querying_sampled_image_fetch_support>>.
```

Comment:
What I don't fully understand is, why do we overload the existing reads instead of adding new functions? It seems less transparent to leave such a decision to the argument type rather than having the user specifically specify whether they want sampling or fetching.

Reply:
We made the decision early on in the extension to have a single `read_image` API to handle every kind of read from both sampled and unsampled images.

While we have had to move away from that a bit e.g. with `read_mipmap` (a decision that was more forced due to ambiguous overloads), w.r.t. changing the naming for fetching vs sampling, we would have to make sure that we are consistent across both sampled and unsampled image reads.

This would mean getting rid of `read_image` entirely. Unsampled images would only have `fetch_image` capabilities, and sampled images would have `sample_image` and `fetch_image` capabilities.

For the time being just keeping `read_image` as an API that encompassed all kinds of reads required much less intrusive changes to the API. However, this is something we are happy to look at and discuss changing in the future.

Reply:
@jbrodman & @gmlueck - Thoughts on this?

Reply:
We're happy to do this.

After discussing with the team, we think the best way to go about making this change is creating a separate small PR, which will deprecate the `read_image` and `read_mipmap` APIs, and create `fetch_image`, `sample_image`, and `sample_mipmap` APIs, for unsampled images, sampled images, and mipmaps respectively. Having a separate NFC PR feels like a better method rather than including it in this, already large, PR, and produces a cleaner git history.

I will draft this PR for now, pending the approval and merging of the above mentioned name change PR. Once that is merged, I will amend this PR to follow the new API naming, with add a `fetch_image` API for standard sampled images.

Reply:
The NFC PR has been created here: https://github.com/intel/llvm/pull/12756

Reply:
Above PR has been merged. I've updated this PR to follow the new API. Un-drafting.